scertecd/scertecd.go

```diff
@@ -110,13 +112,86 @@ func (s *Server) lazyInit() {
 	})
 }
 
+// updateDynamicDomainList updates Server's knowledge of additional
+// domains, dynamically fetched from setec.
+func (s *Server) updateDynamicDomainList(ctx context.Context) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	sec, err := s.SetecClient.GetIfChanged(ctx, s.Prefix+s.DynamicDomainsKey, s.lastDynVers)
+	if err != nil {
+		switch {
+		case errors.Is(err, api.ErrNotFound):
+			s.Logf("additional domains key %q not present on setec, skipping updating the domains list", s.DynamicDomainsKey)
+			return nil
+		case errors.Is(err, api.ErrValueNotChanged):
+			return nil
+		default:
+			return err
+		}
+	}
+	s.lastDynVers = sec.Version
+
+	// Build a map describing the set of additional domains,
+	// and if we saw them represented in dts.
+	addlDomains := make(map[string]bool)
+	for _, domain := range strings.Split(string(sec.Value), ",") {
+		domain = strings.TrimSpace(domain)
+		if domain == "" {
+			continue
+		}
+		addlDomains[domain] = false
+	}
+
+	dts := s.dts[:0:0]
```

Comment:
Why clip? If we don't want to reuse the storage we can as well start with an empty slice.

```go
var dts []domainAndType
```

(I didn't notice whether we are aliasing with someone outside the lock, but if that's a concern we shouldn't bother trying to be clever IMO)